memory/iter.go

```diff
@@ -22,9 +22,6 @@ type iterator struct {
 	// The key at which this iterator ends, inclusive.
 	end []byte
 
-	// If this is true, `start` is inclusive, else `start` is exclusive.
-	isStartInclusive bool
```

Comment:
@fredcarle Sorry as I pushed back on doing this in an earlier PR as I thought this would be more involved than this and require cleanup in badger, however the badger code that I thought was trying to do this was trying to do something else instead and this change was fairly easy.